markdown/assignments/pool_k/tasks/bad_smells_und_refactoring.md

```diff
@@ -0,0 +1,21 @@
+### Bad Smells und Refactoring
+
+In der Vorgabe finden Sie unübersichtlichen und schlecht strukturierten Code. 
+Ihre Aufgabe ist es, für diesen ein Refactoring durchzuführen.
+Ergänzend zu der Übersicht aus der Vorlesung, finden sie unter 
+[Refactoring Guru](https://refactoring.guru/refactoring/techniques) eine erweiterte 
+Auflistung der gängigen Refactoring-Techniken.
+
+Es sollen mindestens die folgenden Refactoring-Techniken angewendet werden. 
+Zeigen Sie in der Abgabe, wo Sie diese verwendet haben. (4P)
```

Comment:
@AMatutat Wir wollten die Punkte eigentlich an der Stelle definieren, wo wir die Aufgabe einbinden. Jetzt haben wir hier allerdings 2x Punkte. Ich sehe zwei Möglichkeiten:

1. Wir lassen die Punkte hier ausnahmesweise drin. 
2. Wir schreiben in der Stelle, wo wir hierher verlinken, die 4+2P hin mit dem Kommentar: "Anwendung Refactoring (4P), Behebung Bad Smells (2P)".

Ich tendiere eher zu (2). Was denkt ihr?

Reply:
Nummer 2 klingt gut

Reply:
no. 2 it is ... to boldy go ... 😎

Reply:
Also fliegen die Punkte dann hier wieder raus? Oder habe ich das falsch verstanden?

Reply:
jupp